### PR TITLE
Promiscuous mode shell command

### PR DESCRIPTION
**Pull request by VivekUppunda**
This set of changes adds a promiscuous mode shell command to shell sample to configure promiscuous mode for the nRF Wi-Fi driver.

### CODE REVIEW

File: samples/wifi/shell/src/wifi_promiscuous_shell.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
```
Comment (krish2718) [inline samples/wifi/shell/src/wifi_promiscuous_shell.c:2]: 2024

File: samples/wifi/shell/src/wifi_promiscuous_shell.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		LOG_ERR("Promiscuous mode %s failed", mode_val ? "enabling" : "disabling");
+	}
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	promisc_cmd,
+	SHELL_CMD_ARG(mode, NULL,
+		      "This command may be used to enable or disable promiscuous mode\n"
+		      "[enable: 1, disable: 0]\n"
```
Comment (krish2718) [inline samples/wifi/shell/src/wifi_promiscuous_shell.c:56]: what about taking the interface index as an arg?

**Review by krish2718 — COMMENTED** [review]
shell code and doc can be in a single commit itself.

File: samples/wifi/shell/src/wifi_promiscuous_shell.c (hunk 2 above)
Comment (VivekUppunda) [reply samples/wifi/shell/src/wifi_promiscuous_shell.c:56]: currently - net_eth_promisc_mode API does not have an ability to configure interface index to the driver. Once that support comes in, we can add the same in the shell sample. It is basically defined as - 
`int net_eth_promisc_mode(struct net_if *iface, bool enable)`

Comment (krish2718) [reply samples/wifi/shell/src/wifi_promiscuous_shell.c:56]: You don't need to pass iface index, you can get iface handle from index and pass it: `iface = net_if_get_by_index`

**Review by krish2718 — APPROVED** [review]

File: samples/wifi/shell/src/wifi_promiscuous_shell.c (hunk 2 above)
Comment (VivekUppunda) [reply samples/wifi/shell/src/wifi_promiscuous_shell.c:56]: Thanks Chaitanya for that input. In the interest of time, I will take that up as a new PR task to update the code.

**Comment by VivekUppunda** [comment]
> shell code and doc can be in a single commit itself.

Merged the same

**Review by richabp — APPROVED** [review]

**Comment by rlubos** [comment]
CLA seems just stuck

**Pull request merged**